fix(mcp): engine-liveness cache hardening - #2454, #2456, #2459 (#3362)
Tr3kkR/Yuzu: docs/ops-runbooks/engine-principal-store-recovery.md

   fully resolved.

## Alert: `YuzuMcpEngineRevalidateStoreUnreachable` (#2367)

**What fired.** `yuzu_server_engine_revalidate_backoff_suppressed_total` moved.
That counter increments ONLY when a liveness re-check for an engine principal
was answered `StoreUnreachable` from the failure backoff — i.e. the store was
already found unreachable moments earlier, and the server is deliberately not
re-asking on every stream tick. It cannot move while the store is healthy, so
this is a true-positive signal about store reachability, not a cache-tuning
metric. The backoff arms only on CONFIRMED unreachability — the store closed,
a query actually ran and failed, or PgPool's own connect-failure breaker is
open (`connect_breaker_open()`, armed only by recent connect failures, never
by pool saturation alone) — so **this alert firing already rules out ordinary
pool saturation with a healthy database**; a bare lease-acquire timeout under
those conditions does not move this counter. One residual edge: if the
transport hangs rather than refusing (a silent firewall drop, a blackholed
route), the breaker only arms once a connect attempt actually times out

> (bounded by `--postgres-connect-timeout`, default 10s) — so the very first

Reviewer: issue(important) - this flag does not exist. `--postgres-connect-timeout` appears exactly once in the tree: this line. The only registered options are `--postgres-dsn` (`main.cpp:327`) and `--postgres-pool-size` (`:332`); the connect timeout is reachable only as a `connect_timeout=` keyword inside the DSN. An on-call following this during a store-availability incident turns a degraded server into a down one. The 10s default is correct.

```suggestion
(bounded by the pool's `connect_timeout`, default 10s - set per-deployment with
```

Author: Fixed - runbook now says the lever is a connect_timeout=<seconds> keyword inside --postgres-dsn, not a separate flag, per your suggestion.
